main/adapters/repository/protocol.go

```diff
@@ -247,6 +283,72 @@ func (p *ExtendedProtocol) IsInitialized(uid uuid.UUID) (initialized bool, err e
 	return true, nil
 }
 
+func (p *ExtendedProtocol) VerifyBackendResponse(requestUPP, responseUPP []byte) (signatureOk bool, chainOk bool, err error) {
+	if !p.verifyNiomonResponse {
+		return false, false, nil
+	}
+
+	// check if backend response is a UPP or something else, like an error message string, for example "Timeout"
+	if !hasUPPHeaders(responseUPP) {
+		return signatureOk, chainOk, fmt.Errorf("response from UBIRCH Trust Service is not a UPP: %q", responseUPP)
+	}
+
+	signatureOk, err = p.verifyBackendResponseSignature(responseUPP)
+	if err != nil {
+		return signatureOk, chainOk, err
+	}
+
+	chainOk, err = p.verifyBackendResponseChain(requestUPP, responseUPP)
+	return signatureOk, chainOk, err
+}
+
+// hasUPPHeaders is a helper function to check if the data starts with the expected UPP headers
+func hasUPPHeaders(data []byte) bool {
+	return bytes.HasPrefix(data, signedUPPHeader) || bytes.HasPrefix(data, chainedUPPHeader)
```

Comment:
What would happen here if data is empty?

Reply:
we would get quite a broad error `response from UBIRCH Trust Service is not a UPP: ""`. do you think, this is acceptable or should i catch this case and log "empty response from UBIRCH Trust Service" ?

Reply:
this message you mentioned would be the body response? or would this message be part of a field in the response and what http code would the user get back?

In general, I think it is good practice to check for empty lists/arrays, so your I like your suggestion.

Reply:
the user would get a `504` response with the JSON response body where the backend response content field would be empty.
The actual error message is only logged internally.

If you think we should add the error message to the response, we I am happy to add this to our roadmap. :)

I added the check for empty input here https://github.com/ubirch/ubirch-client-go/pull/117/commits/d0293611fef1ded51d26bd2973e338a28c8dc2a1

Reply:
Thanks for your reply. I would have one more final question now that we are on this: This error would only get triggered when timing out? I am only curious about the 504.

Reply:
sorry, my bad. i meant `502` not `504`

Reply:
Understood, I get the picture now. Thanks.